uniffi_bindgen/src/bindings/gecko/templates/SharedHeaderTemplate.h

```diff
@@ -0,0 +1,794 @@
+// This file was autogenerated by some hot garbage in the `uniffi` crate.
+// Trust me, you don't want to mess with it!
+
+#ifndef mozilla_dom_{{ ci.namespace()|class_name_webidl }}_Shared
+#define mozilla_dom_{{ ci.namespace()|class_name_webidl }}_Shared
+
+#include <functional>
+
+#include "nsTArray.h"
+#include "prnetdb.h"
+
+#include "mozilla/Casting.h"
+#include "mozilla/CheckedInt.h"
+#include "mozilla/ErrorResult.h"
+#include "mozilla/Utf8.h"
+
+#include "mozilla/dom/BindingDeclarations.h"
+#include "mozilla/dom/Record.h"
+#include "mozilla/dom/{{ ci.namespace()|class_name_webidl }}Binding.h"
+
+extern "C" {
+
+struct RustBuffer {
+  int64_t mLen;
```

Comment:
@rfkelly was talking about changing buffer lengths to `int32_t`.

Reply:
In fact this chance has now landed on the `main` branch.

Reply:
Yep, the length is now `int32_t` and there's also a new capacity field.